src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArr = [];
+
+  resultArr.push(Object.assign({}, state));
+
+  for (let i = 0; i < actions.length; i++) {
```

Comment:
u do not need indexes here, use for of loop, to make your code more readable

Reply:
agree, rewrite your code with for of loop